### ISSUE TITLE
[Bug][CircleCi] Negative workflow duration

### ISSUE THREAD

**Issue by The0Danktor** [apache/devlake#8229]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar issues.

### What happened

Will collecting and transforming data the crashes on the subtask extractJobs

this happens because for some reason a workflow stopped before it started causing it to have a negative durration

and the field is an bigint unsigned

time="2024-12-03 09:07:05" level=error msg=" [pipeline service] [pipeline #3] [task #16] subtask extractJobs ended unexpectedly\n\tWraps: (2) error adding result to batch (500)\n\tWraps: (3) Error 1264 (22003): Out of range value for column 'duration_sec' at row 262 (500)\n\tWraps: (4) Error 1264 (22003): Out of range value for column 'duration_sec' at row 262\n\tError types: (1) *hintdetail.withDetail (2) *hintdetail.withDetail (3) *hintdetail.withDetail (4) *mysql.MySQLError"

### What do you expect to happen

that it would work even with a negative duration
maby by changing them to positive or changing the field type in the database

### How to reproduce

import a workflow in circleci that has a negative duration

### Version

v1.0.1

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

**Comment by klesh** [comment]
Thanks for reporting the issue.

**Comment by JFGHT** [comment]
Workaround:

```sql
ALTER TABLE _tool_circleci_jobs MODIFY COLUMN duration_sec BIGINT DEFAULT NULL;
ALTER TABLE _tool_circleci_workflows MODIFY COLUMN duration_sec BIGINT DEFAULT NULL;
ALTER TABLE _tool_circleci_pipelines MODIFY COLUMN duration_sec BIGINT DEFAULT NULL;
ALTER TABLE cicd_pipelines MODIFY COLUMN duration_sec BIGINT DEFAULT NULL;
ALTER TABLE cicd_tasks MODIFY COLUMN duration_sec BIGINT DEFAULT NULL;
ALTER TABLE cicd_deployments MODIFY COLUMN duration_sec BIGINT DEFAULT NULL;
ALTER TABLE cicd_deployment_commits MODIFY COLUMN duration_sec BIGINT DEFAULT NULL;
```

**Comment by varsis** [comment]
@klesh Can this be reopened? I'm experiencing this issue on my instance.

**Comment by klesh** [comment]
@varsis Sure, did you try @JFGHT 's workaround ?

**Comment by varsis** [comment]
@klesh Yes the query should fix the issue, however I would like to implement a proper fix for this. (For myself and others)

**Comment by zahorniak** [comment]
Hey @klesh, 
We are using version v1.0.3-beta9@1410838 and I can confirm that workaround from this [comment](https://github.com/apache/incubator-devlake/issues/8229#issuecomment-3744594835) worked. Only one minor suggestion: it seems that column `duration_sec` no longer exists in table `_tool_circleci_pipelines`.

**Comment by klesh** [comment]
@varsis  Hi, are you still interested in implementing the fix?

**Comment by varsis** [comment]
> @varsis  Hi, are you still interested in implementing the fix?

Yes I am. I'll see if I can squeeze in sometime this weekend
